config/decoder.go

```diff
@@ -51,24 +52,273 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
-func convertValueFromStruct(value interface{}, targetType reflect.Type, fieldType reflect.Type, fieldValue reflect.Value) error {
+func convertSignedInts(src interface{}, dst *reflect.Value) error {
+	switch t := src.(type) {
+	case int:
+		v := int64(t)
+		if !dst.OverflowInt(v) {
+			dst.SetInt(v)
+			return nil
+		}
+	case uint:
```

Comment:
It does, but then 't' becomes an 'interface{}' and I can't convert to 'int64' easily.